doc/lua_api.txt

```diff
@@ -2590,6 +2591,37 @@ An `InvRef` is a reference to an inventory.
 * `get_location()`: returns a location compatible to `minetest.get_inventory(location)`
     * returns `{type="undefined"}` in case location is not known
 
+### `AreaStore`
+A fast access data structure to store areas, and find areas near a given position or area.
+Every area has a `data` string attribute to store additional information.
```

Comment:
You can't use raw integers with Lua 5.1  They finally come with 5.3 which will probably never get LuaJIT support.

I think ShadowNinja is talking about practicality from a Lua API perspective.  While it's very easy to store and spit back raw data in C++, this adds an additional step for Lua.  On the other hand, it's harder to serialize a raw Lua table and reconstruct it when needed in C++.

You can solve this by associating userdata on a per-ID basis on the Lua side.  Maybe define an API wrapper that automatically stores this for you?